qcodes/tests/dataset/helper_functions.py

```diff
@@ -1,18 +1,24 @@
+from __future__ import annotations
+
 from functools import reduce
 from operator import mul
-from typing import Dict, List, Optional, Sequence, Tuple, Union
+from typing import TYPE_CHECKING, Dict, List, Optional, Sequence, Tuple, Union
```

Comment:
```suggestion
from typing import TYPE_CHECKING, List, Optional, Sequence, Tuple, Union
```

Reply:
Looks like `from typing import TYPE_CHECKING, Sequence` would be enough